src/store/modules/order/mutations.ts

```diff
@@ -8,13 +8,10 @@ const mutations: MutationTree <OrderState> = {
     state.purchaseOrders.total = payload.total;
   },
   [types.ORDER_CURRENT_UPDATED](state, payload) {
-    if (payload.order) {
-      state.current.items = payload.order
-      state.current.orderId = payload.order[0]?.orderId;
-      state.current.externalOrderId = payload.order[0]?.externalOrderId;
-    } else {
-      state.current = payload;
-    }
+    state.current.items = payload.items
```

Comment:
Why not just `state.current = payload;`?

Reply:
Yes Sir, we can do this as well but I thought destructuring might be a better solution.

Reply:
We are updating the current it has 4 keys currently, it may increase and thus make it compulsory for the action to pass. We could update specific property in action and pass current for update, this way it would be easy to handle specific property update

Reply:
Could you please share inputs why destructuring might be a better solution?

Reply:
Sir, your suggestions are better so I have updated the code because instead of destructuring the whole object in mutation we can give a restriction to actions to pass exact 4 properties every time which I have implemented already.